platform/posix/transport/src/openssl_posix.c

```diff
@@ -607,12 +606,21 @@ int32_t Openssl_Recv( NetworkContext_t * pNetworkContext,
     int32_t bytesReceived = 0;
     int sslError = 0;
 
-    /* SSL read of data. */
-    bytesReceived = ( int32_t ) SSL_read( pNetworkContext->pSsl,
-                                          pBuffer,
-                                          bytesToRecv );
+    if( pNetworkContext->pSsl != NULL )
```

Comment:
Is this necessary now that line 584 has been deleted?  Same comment at lines 623 & 650.

Reply:
Is good to have the NULL check for cases when `OpenSSL_Send` and `Openssl_Recv` is called with invalid `pNetworkContext` parameter so that the functions do not Seg Fault

Reply:
That makes sense, but why isn't this method validating `pNetworkContext` before de-referencing it to check `pNetworkContext->pSsl`?

Reply:
That's a good point. Will incorporate the hygiene in this PR.